### ISSUE TITLE
Support passing dm objects to dm()

### ISSUE THREAD

**Issue by krlmlr** [cynkra/dm#826]
Similarly to the way we can pass tibbles to `tibble()` and `mutate()` .

Does this make `dm_bind()` obsolete?

**Comment by moodymudskipper** [comment]
Maybe it helps to think about the differences between `tibble()`, `mutate()` and `bind_cols()` ?

* `bind_cols()` is not data masking and `bind_cols(cars, cars)` will create a 4 col data frame
* `mutate(cars, cars)` will create a 2 col data frame
* `tibble(cars, cars)` fails because of repeated col names

`dm_bind(dm(cars, iris), dm(cars = mtcars))` fails, maybe `dm(dm(cars, iris), dm(cars = mtcars))` would not fail and would overwrite `cars` with the rightmost definition ?

**Comment by krlmlr** [comment]
`tibble()` and `mutate()` now unpack:

``` r
tibble::tibble(cars)
#> # A tibble: 50 × 2
#>    speed  dist
#>    <dbl> <dbl>
#>  1     4     2
#>  2     4    10
#>  3     7     4
#>  4     7    22
#>  5     8    16
#>  6     9    10
#>  7    10    18
#>  8    10    26
#>  9    10    34
#> 10    11    17
#> # … with 40 more rows
```

<sup>Created on 2022-02-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by krlmlr** [comment]
And you're right about the other examples:

``` r
cars |>
  dplyr::mutate(cars)
#>    speed dist
#> 1      4    2
#> 2      4   10
#> 3      7    4
#> 4      7   22
#> 5      8   16
#> 6      9   10
#> 7     10   18
#> 8     10   26
#> 9     10   34
#> 10    11   17
#> 11    11   28
#> 12    12   14
#> 13    12   20
#> 14    12   24
#> 15    12   28
#> 16    13   26
#> 17    13   34
#> 18    13   34
#> 19    13   46
#> 20    14   26
#> 21    14   36
#> 22    14   60
#> 23    14   80
#> 24    15   20
#> 25    15   26
#> 26    15   54
#> 27    16   32
#> 28    16   40
#> 29    17   32
#> 30    17   40
#> 31    17   50
#> 32    18   42
#> 33    18   56
#> 34    18   76
#> 35    18   84
#> 36    19   36
#> 37    19   46
#> 38    19   68
#> 39    20   32
#> 40    20   48
#> 41    20   52
#> 42    20   56
#> 43    20   64
#> 44    22   66
#> 45    23   54
#> 46    24   70
#> 47    24   92
#> 48    24   93
#> 49    24  120
#> 50    25   85
```

<sup>Created on 2022-02-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by krlmlr** [comment]
Anyway, no need to take action right now. Just an idea.

**Comment by krlmlr** [comment]
Accidentally implemented in 1.0.0:

``` r
options(conflicts.policy = list(warn = FALSE))
library(dm)

dm(dm(x = tibble(a = 1)))
#> ── Metadata ────────────────────────────────────────────────────────────────────
#> Tables: `x`
#> Columns: 1
#> Primary keys: 0
#> Foreign keys: 0
```

<sup>Created on 2023-08-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
